ui/lib/kv/addon/components/page/secret/version-history.hbs

```diff
@@ -1,11 +1,102 @@
-<KvPageHeader @breadcrumbs={{@breadcrumbs}} @model={{@model}} @pageTitle={{@model.name}}>
+<KvPageHeader @breadcrumbs={{@breadcrumbs}} @pageTitle={{@path}}>
   <:tabLinks>
     <LinkTo @route="secret.details" data-test-secrets-tab="Secret">Secret</LinkTo>
     <LinkTo @route="secret.metadata.index" data-test-secrets-tab="Metadata">Metadata</LinkTo>
-    <LinkTo @route="secret.metadata.versions" data-test-secrets-tab="Metadata">Version History</LinkTo>
-    <LinkTo @route="secret.metadata.diff" data-test-secrets-tab="Metadata">Version Diff</LinkTo>
+    <LinkTo @route="secret.metadata.versions" data-test-secrets-tab="Version History">Version History</LinkTo>
+    <LinkTo @route="secret.metadata.diff" data-test-secrets-tab="Version Diff">Version Diff</LinkTo>
   </:tabLinks>
-
 </KvPageHeader>
+{{#if @metadata.canReadMetadata}}
+  {{#each @metadata.sortedVersions as |versionData|}}
+    <LinkedBlock
+      class="list-item-row"
+      @params={{array "vault.cluster.secrets.backend.kv.secret.details" @metadata.path}}
+      @queryParams={{hash version=versionData.version}}
+      data-test-version-linked-block={{versionData.version}}
+    >
+      <div class="level is-mobile">
+        <div class="is-grid is-grid-3-columns is-three-fourths-width">
+          {{! version number and icon }}
+          <div class="align-self-center">
+            <Icon @name="history" class="has-text-grey" data-test-version />
+            <span class="has-text-weight-semibold has-text-black">
+              Version
+              {{versionData.version}}
+            </span>
+          </div>
+          {{! icons }}
+          <div class="align-self-center" data-test-icon-holder={{versionData.version}}>
+            {{#if (loose-equal versionData.version @metadata.currentVersion)}}
+              <div>
+                <span class="has-text-success is-size-8 is-block">
+                  <Icon @name="check-circle-fill" />Current
+                </span>
+              </div>
+            {{/if}}
+            {{#if versionData.deletion_time}}
```

Comment:
does this mean a secret could have a grey check and a destroyed icon?

Reply:
TIL: the answer is yes. The API will return a `deletion_time` and a `destroyed === true`. While we don't allow a secret that has been deleted to be destroyed in the UI, the API does not seem to have the same restriction.

![image](https://github.com/hashicorp/vault/assets/6618863/29e01de8-eb60-4406-8cf1-a59ff36fa1d4)

Reply:
![image](https://github.com/hashicorp/vault/assets/6618863/65b72f95-55f3-411b-b4ba-2e4ba1b50ef3)

Reply:
As an aside, this messes with the empty state messages on the secret details view and the icons on the version drop down. For example, it will show a deleted icon, but the empty state will say "Destroyed".

Reply:
Updated to Destroyed is on top and Deleted on bottom if they both exist.